src/index/hnsw/faiss_hnsw.cc

```diff
@@ -2053,7 +2290,7 @@ class BaseFaissRegularIndexHNSWPRQNodeTemplate : public BaseFaissRegularIndexHNS
 };
 
 //
-KNOWHERE_SIMPLE_REGISTER_DENSE_FLOAT_ALL_GLOBAL(FAISS_HNSW_FLAT, BaseFaissRegularIndexHNSWFlatNodeTemplate,
+KNOWHERE_SIMPLE_REGISTER_DENSE_FLOAT_ALL_GLOBAL(FAISS_HNSW_FLAT, BaseFaissRegularIndexHNSWFlatNodeTemplateWithSearchFallback,
```

Comment:
This name needs to be replaced with "HNSW" so that all HNSW requests will be forwarded here.

Reply:
would it cause conflicts with an existing registration for hnswlib's implementation?

Reply:
> would it cause conflicts with an existing registration for hnswlib's implementation?

You must take this approach; otherwise, faiss_hnsw_flat won’t be called, and we don’t plan to introduce a new name. You can rename the original HNSW as HNSW_DEPRECATED（I’ll work on handling CARDINAL on my end）